###### ASSIGNMENTS

##### Assignment-I

1. Write a python program to check if a string is palindrome or not?
<br> print the result as True/False for a list of strings

In [2]:
String1=str(input("enter a string:"))
String2=String1[::-1]
if(String1 == String2):
    print("The given string is a Palindrome")
else:
    print("The given string is not a Palindrome")

enter a string:
The given string is a Palindrome


2. Write a Python program to count number of empty substrings of a given string?
<br> Program should be able to take input at runtime

In [4]:
String1=str(input("enter a string:"))
result=String1.count(" ")
print("number of empty substrings {}".format(result))

enter a string:
number of empty substrings 0


3. Write a python program to retrieve symbols from a given string?

In [5]:
import re
text = input (print("Enter Text"))
result = re.findall('[^0-9a-zA-Z" "]+',text)
print(result)

Enter Text
Noneruhi@arshi$jhy
['@', '$']


4. Write a python program to count the occurence of all vowels? 
<br> For instance in the string "Hello World"
<br> e -1 
<br> o -2
<br> Input should be fetched at runtime.

In [ ]:
x = str (input (print("Enter a text")))

print ( "a - :", x.count("a"))
print ( "e - :", x.count("e"))
print ( "i - :", x.count("i"))
print ( "o - :", x.count("o"))
print ( "u - :", x.count("u"))

5. Generate Fibonacci Series till 100?


In [ ]:
a = 0
b = 1
c = 0
if nterm <= 0:
    print("enter positve num")
elif n term == 1:
    print ("fibonacci series upto",nterm,":")
    while c < nterm:
        print(a,end=' , ')
        nterm = a + b
        a = b
        b = nterm
        c += 1
    


6. print True to words having more vowels than consanants provided a list of words?
["abc","aba"]
FALSE, TRUE


In [ ]:
import re
strings=["hellow","ath","world","abiocudie"]
def function1(String1):
    result=re.findall("[aeiou]",String1)
    result1=re.findall("[^aeiou]",String1)
    print(result,result1)
    a=len(result)
    b=len(result1)
    print(a,b)
    return a>b
    
print(list(map(function1,strings)))

7. Write a program which can compute the factorial of a given number?

In [ ]:
def factorial(n):
    return 1 if(n==0 or n==1) else n*factorial(n-1)

number=int(input("enter a number:"))
print("factorial :", factorial(number))
           

8. write a program to find the frequency of all words from a paragraph, and create a dictionary of all words with word being the key and the count being it's value?
<br> frequency should be considered ignoring word contained in another word

<br>"python runs very fast, it's runtime is very lesser, eventhough the program is running on an interpreter which has a higher runtime than an compiler based programming languages."

<br> ** here program count is - 1 and it shouldn't be counted in the programming which will make it to 2 **


In [3]:
import re
paragraph="""Define a function which can generate a dictionary where the keys are numbers between 1 and 20 (both included)
and the values should be a dictionary with square and cube as keys and with values being it's square and cube. 
the function should print the dictionary in pretty format """
result=re.findall("\w+",paragraph)
#print(result)
print([{x:result.count(x)} for x in result])

[{'Define': 1}, {'a': 3}, {'function': 2}, {'which': 1}, {'can': 1}, {'generate': 1}, {'a': 3}, {'dictionary': 3}, {'where': 1}, {'the': 4}, {'keys': 2}, {'are': 1}, {'numbers': 1}, {'between': 1}, {'1': 1}, {'and': 5}, {'20': 1}, {'both': 1}, {'included': 1}, {'and': 5}, {'the': 4}, {'values': 2}, {'should': 2}, {'be': 1}, {'a': 3}, {'dictionary': 3}, {'with': 2}, {'square': 2}, {'and': 5}, {'cube': 2}, {'as': 1}, {'keys': 2}, {'and': 5}, {'with': 2}, {'values': 2}, {'being': 1}, {'it': 1}, {'s': 1}, {'square': 2}, {'and': 5}, {'cube': 2}, {'the': 4}, {'function': 2}, {'should': 2}, {'print': 1}, {'the': 4}, {'dictionary': 3}, {'in': 1}, {'pretty': 1}, {'format': 1}]


9. Define a function which can generate a dictionary where the keys are numbers between 1 and 20 (both included)
<br>
and the values should be a dictionary with square and cube as keys and with values being it's square and cube.
<br>
the function should print the dictionary in pretty format?
<br>
{1:{"square":1, "cube": 1},2:{"square":4, "cube":8},3:{"square":9, "cube":27},..... and so on until 20

In [2]:
square=lambda s:s**2
cube=lambda c:c**3

print([{x:{"square":square(x),"cube":cube(x)}} for x in range(1,21)])

[{1: {'square': 1, 'cube': 1}}, {2: {'square': 4, 'cube': 8}}, {3: {'square': 9, 'cube': 27}}, {4: {'square': 16, 'cube': 64}}, {5: {'square': 25, 'cube': 125}}, {6: {'square': 36, 'cube': 216}}, {7: {'square': 49, 'cube': 343}}, {8: {'square': 64, 'cube': 512}}, {9: {'square': 81, 'cube': 729}}, {10: {'square': 100, 'cube': 1000}}, {11: {'square': 121, 'cube': 1331}}, {12: {'square': 144, 'cube': 1728}}, {13: {'square': 169, 'cube': 2197}}, {14: {'square': 196, 'cube': 2744}}, {15: {'square': 225, 'cube': 3375}}, {16: {'square': 256, 'cube': 4096}}, {17: {'square': 289, 'cube': 4913}}, {18: {'square': 324, 'cube': 5832}}, {19: {'square': 361, 'cube': 6859}}, {20: {'square': 400, 'cube': 8000}}]


10. write a program to find duplicates in a list?

In [ ]:
example_list=[2,3,4,5,1,2,4,6,7]
a=[]
for x in example_list:
    if x not in a:
        if(example_list.count(x))>1:
            a.append(x)
print(a) 

for x in a:
    print("index of {} is {}".format(x,example_list.index(x)))


11. write a program to list out the duplicate indexes in list, the first occurence being non duplicate?

12. Create a generator which can take an integer input and generate a list of numbers having integer squareroots and cuberoots?
like
<br> 1 --  1*1 || 1*1*1
<br> 64 --  8*8 || 4*4*4
<br> 729 --  27*27 || 9*9*9 


<br>
<br>
<br>
if generator is not taken please work on that using block and conditional statements

In [1]:
import math

def func1(val):
    for x in range(val):
        y=math.sqrt(x)
        z=x**(1/3)
        if y.is_integer():
            print("square root of {} having integer value is {}" .format(x,int(y)))
        if z.is_integer():
            print("cube root of {} having integer value is {}" .format(x,int(z)))
               

value=int(input("enter a interger:"))
func1(value)
        

enter a interger:3
square root of 0 having integer value is 0
cube root of 0 having integer value is 0
square root of 1 having integer value is 1
cube root of 1 having integer value is 1
